xkcdpass/xkcd_password.py

```diff
@@ -203,8 +202,12 @@ def choose_words(wordlist, numwords):
     """
     Choose numwords randomly from wordlist
     """
-
-    return [rng().choice(wordlist) for i in xrange(numwords)]
+    assert len(wordlist) > 0
+   
+    if len(wordlist) >= numwords:
+        return random.sample(wordlist, numwords)  # ensures no duplicates
```

Comment:
Diceware etc assume truly random word generation, hence `choice` vs `sample` - duplicate words are unlikely with any decent wordlist in any case. I would prefer raising an error in `validate_options` if the wordlist is shorted than `numwords` (as this would be quite insecure).